tsconfig.production.json

```diff
@@ -14,11 +14,11 @@
     "noUnusedParameters": true,
     "outDir": "lib",
     "pretty": true,
-    "rootDir": ".",
+    "resolveJsonModule": true,
+    "rootDir": "./src",
     "sourceMap": true,
     "strict": true,
     "strictNullChecks": true
   },
-  "include": ["./src/**/*.ts"],
   "exclude": ["test/**/*.ts"]
```

Comment:
We should remove it.

Reply:
Why?

Reply:
Because you changed the `rootDir`.

Reply:
if I remove it I get `TS6059` errors for the test files:
```
error TS6059: File 'eslint-plugin/test/index.spec.ts' is not under 'rootDir' 'eslint-plugin/src'. 'rootDir' is expected to contain all source files.

error TS6059: File 'eslint-plugin/test/rules/no-single-line-comment-block.spec.ts' is not under 'rootDir' 'eslint-plugin/src'. 'rootDir' is expected to contain all source files.

error TS6059: File 'eslint-plugin/test/rules/no-template-literals-without-expression.spec.ts' is not under 'rootDir' 'eslint-plugin/src'. 'rootDir' is expected to contain all source files.
```

So it is necesary aas not all the `.ts` files are under `src`.

Reply:
Note that this is `tsconfig.PRODUCTION.json` file. There's `tsconfig.json`. I don't know where do you get these errors, if it's tests, local language server, linter, ... - but they all use `tsconfig.json`.

If we're getting these errors for the build task, something is wrong - these files should never be imported.

Reply:
from the docs:
>If no 'files' or 'include' property is present in a tsconfig.json, the compiler defaults to including all files in the containing directory and subdirectories except those specified by 'exclude'.

So I guess I can either leave this as it is or actually change it with:
```JSON
"include": ["./src"]
```
Either one works.

Reply:
Or better, files? With just the entrypoint?

Reply:
I tried:
```JSON
"files": ["./src/index.ts"]
```
But then it only builds for that file, it does not include any other rule file. So I will just keep the `include` as stated above.

Reply:
fixed with: https://github.com/ridedott/eslint-plugin/pull/30/commits/39a99c8f24ae4b9c71e569dd4f761fe4508058a9